plugin/register.go

```diff
@@ -0,0 +1,57 @@
+package plugin
+
+import (
+	"fmt"
+	"github.com/mitchellh/cli"
+	"reflect"
+	"strings"
+	"trellis-cli/cmd"
+)
+
+func Register(c *cli.CLI, searchPaths []string, validPluginFilenamePrefixes []string) {
+	coreNamespaces := namespacesFor(reflect.ValueOf(c.Commands))
+	pluginFinder := newFinder(validPluginFilenamePrefixes, searchPaths, coreNamespaces)
+	plugins := pluginFinder.find()
+	pluginNamespaces := namespacesFor(reflect.ValueOf(plugins))
+
+	// Register plugin commands.
+	for name, bin := range plugins {
+		pluginCommand := cmd.NewPassthroughCommand(name, bin, c.Args)
+		c.Commands[name] = func() (cli.Command, error) {
+			return pluginCommand, nil
+		}
+	}
+
+	// Separate plugin commands from core command lists
+	c.HiddenCommands = append(c.HiddenCommands, pluginNamespaces...)
+	// Append plugin command list to help text.
+	c.HelpFunc = helpFunc(c.Name, pluginNamespaces)
+}
+
+func namespacesFor(v reflect.Value) []string {
+	m := v.MapKeys()
+
+	var namespaces []string
+	for _, v := range m {
+		name := fmt.Sprint(v)
+		namespace := strings.Split(name, " ")[0]
+		namespaces = append(namespaces, namespace)
+	}
+
+	return unique(namespaces)
+}
+
+// unique de-duplicates a given slice of strings without
+// sorting or otherwise altering its order in any way.
+func unique(values []string) []string {
```

Comment:
```go
func unique(values []string) (newValues []string) {
```

Reply:
done in [`c241027` (#144)](https://github.com/roots/trellis-cli/pull/144/commits/c241027becb2641840f9245a88baa589cc235925)